internal/app/http-adapter/apicontroller/methods.go

```diff
@@ -0,0 +1,154 @@
+	var task model.Task
+
+	err := json.NewDecoder(r.Body).Decode(&task)
+	if err != nil {
+		http.Error(w, `{"error":"Failed to decode JSON"}`, http.StatusBadRequest)
+		return
+	}
+
+	// поле title не указано
+	if task.Title == "" {
+		http.Error(w, `{"error":"Empty title field"}`, http.StatusBadRequest)
+		return
+	}
+
+	id, err := ctr.taskService.AddTask(task)
+	if err != nil {
+		http.Error(w, `{"error":"`+err.Error()+`"}`, http.StatusInternalServerError)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	json.NewEncoder(w).Encode(map[string]interface{}{"id": id})
+}
+
+// Получение списка задач с фильтром (GET)
+func (ctr Controller) GetTasks(w http.ResponseWriter, r *http.Request) {
+	search := r.URL.Query().Get("search")
+
+	tasks, err := ctr.taskService.GetTasks(search)
+	if err != nil {
+		http.Error(w, `{"error":"`+err.Error()+`"}`, http.StatusInternalServerError)
+		return
+	}
+
+	if len(tasks) == 0 {
+		tasks = []model.Task{}
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	json.NewEncoder(w).Encode(map[string]interface{}{"tasks": tasks})
+}
+
+// Получение информации о задаче (GET)
+func (ctr Controller) GetTaskByID(w http.ResponseWriter, r *http.Request) {
+	id := r.URL.Query().Get("id")
+
+	task, err := ctr.taskService.GetTaskByID(id)
+	if err != nil {
+		http.Error(w, `{"error":"`+err.Error()+`"}`, http.StatusBadRequest)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	json.NewEncoder(w).Encode(task)
+}
+
+// Обновление задачи (PUT)
+func (ctr Controller) Update(w http.ResponseWriter, r *http.Request) {
+	var task model.Task
+
+	err := json.NewDecoder(r.Body).Decode(&task)
+	if err != nil {
+		http.Error(w, `{"error":"Failed to decode JSON"}`, http.StatusBadRequest)
+		return
+	}
+
+	err = ctr.taskService.Update(task)
+	if err != nil {
+		if err.Error() == "задача не найдена" {
+			http.Error(w, `{"error":"Task not found"}`, http.StatusNotFound)
+		} else {
+			http.Error(w, `{"error":"`+err.Error()+`"}`, http.StatusInternalServerError)
+		}
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	json.NewEncoder(w).Encode(map[string]interface{}{})
+}
+
+// Удаление задачи (DELETE)
+func (ctr Controller) Delete(w http.ResponseWriter, r *http.Request) {
+	id := r.URL.Query().Get("id")
+
+	err := ctr.taskService.Delete(id)
+	if err != nil {
+		http.Error(w, `{"error":"`+err.Error()+`"}`, http.StatusBadRequest)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	json.NewEncoder(w).Encode(map[string]interface{}{})
+}
+
+// Обработчик, который делает задачу выполненной (POST)
+func (ctr Controller) MarkTaskDone(w http.ResponseWriter, r *http.Request) {
+	id := r.URL.Query().Get("id")
+	if id == "" {
+		http.Error(w, `{"error":"Missing task ID"}`, http.StatusBadRequest)
+		return
+	}
+
+	now := time.Now()
+
+	err := ctr.taskService.MarkTaskDone(id, now)
+	if err != nil {
+		if err.Error() == "не удалось получить задачу: задача не найдена" {
```

Comment:
Так лучше не делать.
Нужно определить, где будут ошибки и, например, в пакете БД в начале сделать глобальную ошибку
`var ErrTaskNotFound = "task not found"`
а тут
```suggestion
		if errors.IS(err, sqlite.ErrTaskNotFound){
```
Тогда будет очень красиво